File: minha-vacina-app/src/models/usuario.ts
import { Campanha } from "./campanha";
import { Municipio } from "./municipio";
import { Permissao } from "./permissao";
import { TokenNotificacao } from "./tokenNotificacao";
import { Vacina } from "./vacina";

export interface Usuario {
  id?: number;
  permissao?: Permissao[];
  nome: string;
  municipio: Municipio;
  dataNascimento: Date;
  email: string;
  senha: string;
  vacinas?: Vacina[];
  campanhas?: Campanha[];
}

File: minha-vacina-front/src/main/java/br/com/minhavacina/domain/Usuario.java
package br.com.minhavacina.domain;

import lombok.Data;

import java.util.Date;
import java.util.List;

@Data
public class Usuario {
    private Integer id;
    private String nome;
    private String email;
    private String senha;
    private Date dataNascimento;
    private Municipio municipio;
    private List<Vacina> vacinas;
    private List<Permissao> permissoes;
}
